ea-2021-04-ndvi-automation.py

```diff
@@ -0,0 +1,683 @@
+# HINT: FOR THIS STEP, leave any rows containing missing values (`NAN`).
+
+# In[9]:
+
+
+# Create dataframe of NDVI including the cleaning data to deal with clouds
+
+# Important: to use the ungraded tests below as a sanity check,
+# name your columns: mean_ndvi and site
+# Don't forget to set date as the index and make the values of type datetime
+
+# YOUR CODE HERE
+
+
+harv_landsat_dir = os.path.join(sites[1],
+                           "landsat-crop")
+harv_landsat_dirs = sorted(glob(os.path.join(harv_landsat_dir, "LC08*")))
+
+
+
+sjer_landsat_dir = os.path.join(sites[0],
+                           "landsat-crop")
+sjer_landsat_dirs = sorted(glob(os.path.join(sjer_landsat_dir, "LC08*")))
+
+
+landsat_dirs_all = harv_landsat_dirs + sjer_landsat_dirs
+
+ndvi_means = []
+harv_dates = []
+site_names = []
+for adir in landsat_dirs_all:
+    band_paths_harv_sjer = sorted(glob(os.path.join(adir, "*band*[4-5].tif")))
+    
+    all_bands_harv_sjer = []
+    for aband in band_paths_harv_sjer:
+        cleaned_band = open_clean_bands(band_path=aband, valid_range=(0, 10000))
+        all_bands_harv_sjer.append(cleaned_band)
+        date = aband[50:58]
+    
+    #CLOUD MASK
+    
+    qa_xr_path_each = glob(os.path.join(adir, "*qa*"))
+    # Clip the cloud mask layer
+    cl_mask = rxr.open_rasterio(qa_xr_path_each[0], masked=True).squeeze()
+    # Get a list of each directory
+    path_each = os.path.join("ndvi-automation", "sites")
+    # Get a list of both site directories
+    sites = glob(path + "/*/")
+    # Open up the shapefile for clipping your landsat data to the study area
+    if adir[22:26]=='HARV':
+        vector_dir = os.path.join(sites[1],
+                          "vector")
+    if adir[22:26]=='SJER':
+        vector_dir = os.path.join(sites[0],
+                          "vector")
+        
+    site_name_each = adir[22:26]
+
+    
+    # Open crop boundary
+    site_boundary_path_all = os.path.join(vector_dir,  site_name_each + "-crop.shp")
+    crop_bound_all = gpd.read_file(site_boundary_path_all)
+
+    
+    
+    
+    ndvi_clean = mask_crop_ndvi(all_bands=all_bands_harv_sjer,
+                            crop_bound=crop_bound_all,
+                            pixel_qa=cl_mask,
+                            vals=vals)
+    
+    mean_ndvi = np.mean(ndvi_clean).values
+    harv_date = pd.to_datetime(date, format='%Y%m%d')
+
+    
+    ndvi_means.append(mean_ndvi)
+    harv_dates.append(harv_date)
+    site_names.append(site_name_each)
+    
+    
+
+all_ndvi_dict = {'site': site_names, 'mean_ndvi': ndvi_means, 'date': harv_dates}
+all_ndvi_mean_df = pd.DataFrame.from_dict(all_ndvi_dict)
+
+all_ndvi_mean_df_date_index = all_ndvi_mean_df.set_index('date')
+
+
+#convert mean_ndvi to floats
+all_ndvi_mean_df_date_index["mean_ndvi"] = all_ndvi_mean_df_date_index["mean_ndvi"].astype(float)
+
+all_ndvi_mean_df_date_index.sort_index(inplace=True)
+
+all_ndvi_mean_df_date_index
```

Comment:
good loop here, perhaps some more descriptive notes for the steps? 

vector names are expressive and easy to follow

Reply:
I agree about including more descriptive notes.